src/components/DocumentRenderer/ScannedDocumentRendererContainer.test.tsx

```diff
@@ -0,0 +1,96 @@
+import sampleDocument from "../../../fixtures/demo-caas.json";
+import { CheckStatus } from "../../constants/verifier";
+import * as useDocumentVerifier from "../../common/hooks/useDocumentVerifier";
+
+jest.mock("../DocumentRenderer/WebViewFrame");
+jest.mock("../../common/navigation");
+jest.mock("../../common/hooks/useDocumentVerifier");
+jest.useFakeTimers();
+
+const mockUseDocumentVerifier = useDocumentVerifier.useDocumentVerifier as jest.Mock;
+
+const whenDocumentIsVerified = (): void => {
+  mockUseDocumentVerifier.mockReturnValue({
+    tamperedCheck: CheckStatus.VALID,
+    issuedCheck: CheckStatus.VALID,
+    revokedCheck: CheckStatus.VALID,
+    issuerCheck: CheckStatus.VALID,
+    overallValidity: CheckStatus.VALID
+  });
+};
+
+describe("ScannedDocumentRendererContainer", () => {
+  // eslint-disable-next-line jest/no-hooks
+  beforeEach(() => {
+    resetNavigation();
+    resetDb();
+  });
+
+  it("should render DocumentRenderer with document from navigation params", () => {
+    expect.assertions(1);
+    setParam("document", sampleDocument);
+    setParam("savable", false);
+    const { queryByTestId } = render(
+      <ScannedDocumentRendererContainer navigation={mockNavigation} />
+    );
+    expect(queryByTestId("mock-web-view-frame")).not.toBeNull();
+  });
+
+  it("should render ScannedDocumentActionSheet with save button for savable document", () => {
+    expect.assertions(2);
+    setParam("document", sampleDocument);
+    setParam("savable", true);
+    const { queryByText } = render(
+      <ScannedDocumentRendererContainer navigation={mockNavigation} />
+    );
+    expect(queryByText("Save")).not.toBeNull();
+    expect(queryByText("Done")).toBeNull();
+  });
+
+  it("should render ScannedDocumentActionSheet with done button for unsavable document", () => {
+    expect.assertions(2);
+    setParam("document", sampleDocument);
+    setParam("savable", false);
+    const { queryByText } = render(
+      <ScannedDocumentRendererContainer navigation={mockNavigation} />
+    );
+    expect(queryByText("Save")).toBeNull();
+    expect(queryByText("Done")).not.toBeNull();
+  });
+
+  it("should save and navigate to document to db when save is pressed", async () => {
+    expect.assertions(6);
+    setParam("document", sampleDocument);
+    setParam("savable", true);
+    whenDocumentIsVerified();
+    const { getByText } = render(
+      <MockDbProvider>
+        <ScannedDocumentRendererContainer navigation={mockNavigation} />
+      </MockDbProvider>
+    );
+    fireEvent.press(getByText("Save"));
+    const insertedDocument = mockInsert.mock.calls[0][0];
+    expect(insertedDocument.id).toBe(
+      "b40e1ebcbf786e9188b79b13162bfb276df4e7c5d7fef8d944e797766ef8e97a"
+    );
+    expect(insertedDocument.isVerified).toBe(true);
+    expect(insertedDocument.document).toStrictEqual(sampleDocument);
+    expect(insertedDocument.created).toBeTruthy(); // eslint-disable-line jest/no-truthy-falsy
```

Comment:
Would it be possible to use `.toHaveProperty` rather than add the eslint exception?

Reply:
Good idea.